File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline
n = int(input())
stack = []
for i in range(n):
    text = input().split()
    if text[0]=='push':
        stack.append(text[1])
    elif text[0]=='pop':
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())
    elif text[0]=='size':
        print(len(stack))
    elif text[0]=='empty':
        if len(stack)==0:
            print(1)
        else:
            print(0)
    elif text[0]=='top':
        if len(stack)==0:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();
        
        int N = sc.nextInt();
        sc.nextLine();

        for (int i = 0; i < N; i++) {
            String input = sc.nextLine();
            String[] parts = input.split(" ");
            String command = parts[0];

            switch (command) {
                case "push":
                    int value = Integer.parseInt(parts[1]);
                    stack.push(value);
                    break;
                case "pop":
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.pop());
                    }
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    System.out.println(stack.isEmpty() ? 1 : 0);
                    break;
                case "top":
                    if (stack.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.peek());
                    }
                    break;
            }
        }
        sc.close();
    }
}